docs/source/howto/data.rst

```diff
@@ -260,6 +260,277 @@ Finding data
+
+.. code-block:: console
+
+      $ verdi group relabel test_group old_group
+      Success: Label changed to old_group
+
+From the Python interface:
+
+.. code-block:: ipython
+
+    In [1]: group = load_group(label='old_group')
+
+    In [2]: group.label = 'another_group'
+
+
+Delete a group
+^^^^^^^^^^^^^^
+From the command line interface:
+
+.. code-block:: console
+
+      $ verdi group delete another_group
+      Are you sure to delete Group<another_group>? [y/N]: y
+      Success: Group<another_group> deleted.
+
+.. important::
+    Any deletion operation related to groups won't affect the nodes themselves.
+    For example if you delete a group, the nodes that belonged to the group will remain in the database.
+    The same happens if you remove nodes from the group -- they will remain in the database but won't belong to the group anymore.
+
+Copy one group into another
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This operation will copy the nodes of the source group into the destination group.
+If the destination group does not yet exist, it will be created automatically.
+
+From the command line interface:
+
+.. code-block:: console
+
+    $ verdi group copy source_group dest_group
+    Success: Nodes copied from group<source_group> to group<dest_group>
+
+From the Python interface:
+
+.. code-block:: ipython
+
+    In [1]: src_group = Group.objects.get(label='source_group')
+
+    In [2]: dest_group = Group(label='destination_group').store()
+
+    In [3]: dest_group.add_nodes(list(src_group.nodes))
+
+
+Examples for using groups
+-------------------------
+
+In this section, we will provide some practical examples of how one can use Groups to structure and organize the nodes in the database.
+
+Example I: grouping structures with a similar property
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To put the structures for which the computed bandgap is higher than ``1.0 eV`` in the group ``promising_structures`` one should:
+
+.. code-block:: python
+
+    # Finding the structures with the bandgap > 1.0.
+    qb = QueryBuilder()
+    qb.append(StructureData,  tag='structure', project='*') # Here we are projecting the entire structure object
+    qb.append(CalcJobNode, with_incoming='structure', tag='calculation')
+    qb.append(Dict, with_incoming='calculation', filters={'attributes.bandgap': {'>': 1.0}})
+
+    # Adding the structures in 'promising_structures' group
+    group = load_group(label='promising_structures')
+    group.add_nodes(q.all(flat=True))
+
+.. note::
+
+    Any node can be included in a group only once and if it is added again, it is simply ignored.
+    This means that add_nodes can be safely called multiple times, and only nodes that weren't already part of the group, will be added.
+
+
+Example II: work with data in the groups
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here is a prototype script that would allow us to submit calculations for the structures belonging to the ``promising_structures`` group.
```

Comment:
Here we demonstrate how to submit calculations for structures that all belong to a group named ``promising_structures``: